Задача 1

Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150.
Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [1]:
import numpy as np
salary = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]

In [2]:
salary_avg = sum(salary)/len(salary)
salary_avg

65.3

In [3]:
np.mean(salary)

65.3

Ответ: среднее арифметическое 65.3

In [4]:
def unbiased_std(data: list):
    avg = sum(data)/len(data)
    summa = 0
    for s in data:
        summa += (s - avg)**2
    return np.sqrt(summa/(len(data)-1))
salary_std = unbiased_std(salary)
salary_std

31.624607341019814

In [5]:
np.std(salary, ddof=1)

31.624607341019814

Ответ: среднее квадратичное отклонение 31.62

In [6]:
def shift_var(data):
    avg = sum(data)/len(data)
    summa = 0
    for s in data:
        summa += (s - avg)**2
    return summa/len(data)
salary_var = shift_var(salary)
salary_var

950.11

In [7]:
np.var(salary)

950.11

Ответ: смещенная 950.11

In [8]:
def unbiased_var(data):
    avg = sum(data)/len(data)
    summa = 0
    for s in data:
        summa += (s - avg)**2
    return summa/(len(data)-1)
salary_un_var = unbiased_var(salary)
salary_un_var

1000.1157894736842

In [9]:
np.var(salary, ddof=1)

1000.1157894736842

Ответ: несмещенная 1000.12

Задача 2

В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

In [10]:
def combi(a,b):
    return np.math.factorial(b)/(np.math.factorial(a)*np.math.factorial(b-a))

In [11]:
P = (combi(2,5)*combi(1,5)*combi(3,7)+
     combi(1,5)*combi(1,3)*combi(2,5)*combi(2,7)+
     combi(2,3)*combi(3,5)*combi(1,7))/(combi(2,8)*combi(4,12))
P

0.3686868686868687

Ответ: 36,86%

Задача 3

На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6.
Найти вероятность того, что выстрел произведен:
a) первым спортсменом
б) вторым спортсменом
в) третьим спортсменом

In [12]:
P_A = 0.9/3+0.8/3+0.6/3
P_A

0.7666666666666666

In [13]:
P_B1A = 0.9/3/P_A
P_B1A

0.391304347826087

Ответ: первым спортсменом 0.39

In [14]:
P_B2A = 0.8/3/P_A
P_B2A

0.3478260869565218

Ответ: вторым спортсменом 0.35

In [15]:
P_B3A = 0.6/3/P_A
P_B3A

0.2608695652173913

Ответ: третьим спортсменом 0.26

Задача 4

В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9.
Студент сдал первую сессию. Какова вероятность, что он учится:
a) на факультете A
б) на факультете B
в) на факультете C

In [16]:
P_A = 0.8/4+0.7/4+0.9/2
P_A

0.825

In [17]:
P_B1A = 0.8/4/P_A
P_B1A

0.24242424242424246

Ответ: на факультете A 24.24%

In [18]:
P_B2A = 0.7/4/P_A
P_B2A

0.21212121212121213

Ответ: на факультете B 21.21%

In [19]:
P_B3A = 0.9/2/P_A
P_B3A

0.5454545454545455

Ответ: на факультете C 54.54%

Задача 5

Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25.
Какова вероятность того, что в первый месяц выйдут из строя:
а) все детали
б) только две детали
в) хотя бы одна деталь
г) от одной до двух деталей

In [20]:
P_all = 0.1*0.2*0.25
P_all

0.005000000000000001

Ответ: все детали 0.5%

In [21]:
P_2 = 0.1*0.2*0.75+0.1*0.8*0.25+0.9*0.2*0.25
P_2

0.08000000000000002

Ответ: только две детали 8%

In [22]:
P_1 = 1 - 0.9*0.8*0.75
P_1

0.45999999999999996

Ответ: хотя бы одна деталь 46%

In [23]:
P_1_2 = P_1 - P_all
P_1_2

0.45499999999999996

Ответ: от одной до двух деталей 45.5%